explorer/ast/value.h

```diff
@@ -1062,7 +1061,9 @@ struct ImplsConstraint : public HashFromDecompose<ImplsConstraint> {
 };
 
 // A constraint that requires an intrinsic property of a type.
-struct IntrinsicConstraint : public HashFromDecompose<IntrinsicConstraint> {
+struct IntrinsicConstraint
+    : public Printable<IntrinsicConstraint,
+                       HashFromDecompose<IntrinsicConstraint>> {
```

Comment:
I believe this is necessary to support `ConstructFromComponents` in `value_transform.h`, which was made to support aggregate initialization of a type with a single CRTP base class, but not a type with multiple of them.

In this case, I think we could add a constructor to `IntrinsicConstraint` instead of modeling it as an aggregate to avoid this chaining of CRTP bases.

Reply:
Done